python/ray/monitor.py

```diff
@@ -111,10 +111,14 @@ def __init__(self, redis_address, redis_port, autoscaling_config):
         self.issue_gcs_flushes = "RAY_USE_NEW_GCS" in os.environ
         self.gcs_flush_policy = None
         if self.issue_gcs_flushes:
-            # For now, we take the primary redis server to issue flushes,
-            # because task table entries are stored there under this flag.
+            # Data is stored under the first data shard, so we issue flushes to
```

Comment:
Let's add a note or todo here that this needs to change when we implement sharding